src/kakarot/execution_context.cairo

```diff
@@ -258,13 +254,18 @@ namespace ExecutionContext {
     func revert(
         self: model.ExecutionContext*, revert_reason: felt*, size: felt
     ) -> model.ExecutionContext* {
-        memcpy(self.return_data, revert_reason, size);
+        alloc_locals;
+        let return_data: felt* = alloc();
+        memcpy(return_data, revert_reason, size);
+        tempvar return_info: model.ReturnInfo* = new model.ReturnInfo(
+            len=size, data=return_data, size=0, offset=0
```

Comment:
here we have many concepts: len, size, offset,
could we maybe find more explicit names for it?

we got:
len=size, size=0 my brain cannot comprehend

Reply:
here I don't understand why we set size and offset to 0

Reply:
maybe adding comments to walk the reader through the process can help

Reply:
its still a draft because i was focused on the structural changes and passing the test suite before making it reader friendly :)